proposal/color-4-new-spaces.md

```diff
@@ -0,0 +1,1790 @@
+  `color.to-space()`, which can be used for manual space conversion. Functions
+  that accept two colors (e.g. `color.mix()`) return a color in the same space
+  as the first color argument.
+* No color function performs gamut-mapping on out-of-gamut channels, except
+  `color.to-gamut()`, which can be used for manual gamut-mapping.
+
+Unfortunately, the legacy `hsl` and `hwb` color spaces are not able to express
+out-of-gamut colors, even with out-of-range channel values, so any conversion
+into those spaces (using `color.to-gamut()` or manipulating colors in those
+spaces) must always require gamut-mapping into the `srgb` gamut. This is
+defined as part of the [CSS Color Level 4][color-4] specification for
+[converting] colors.
+
+## Definitions
+
+### Color
+
+> Note that channel values are stored as specified, maintaining precision where
+> possible, even when the values are out-of-gamut for the [known color space].
+
+A *color* is an object with several parts:
+
+* A *color space* that is either a [known color space] or an unquoted string.
+
+* An ordered list of numeric *channel* values.
+
+* A floating-point number *alpha* value between `0-1`, inclusive.
+
+  > While it's valid to specify numbers outside this range, they are
+  > meaningless, and can be clamped by input functions when generating a color.
+
+[known color space]: #known-color-space
+
+### Legacy Color
+
+> Both Sass and CSS have similar legacy behavior that relies on all colors
+> being interchangeable as part of a shared `srgb` color space. While the new
+> color formats will opt users into new default behavior, some legacy color
+> formats behave differently for the sake of backwards-compatibility.
+
+Colors in the `rgb`, `hsl`, or `hwb` [color spaces](#known-color-space) are
+considered *legacy colors*. The output of a legacy color is not required to
+match the input color space, and several color functions maintain legacy
+behavior when manipulating legacy colors.
+
+> This includes colors defined using the CSS color names, hex syntax, `rgb()`,
+> `rgba()`, `hsl()`, `hsla()`, or `hwb()` -- along with colors that are
+> manually converted into legacy color spaces.
+
+### Known Color Space
+
+Sass colors are stored as part of a known color space. Each space has a name
+and an ordered list of associated channels. Each channel has a name and
+position index (1-indexed) defined by the space and the order of channels in
+that space, and a number value with units matching those allowed by the space.
+Space and channel names match unquoted strings, ignoring case. They are always
+emitted as unquoted lowercase strings by inspection functions.
+
+Values outside a _bounded gamut_ range are valid, and remain un-clamped, but
+are considered _out of gamut_ for the given color space. If the channel is
+bounded, or has a percentage mapping with a lower-boundary of zero, then the
+channel is considered _scalable_.
```

Comment:
This term "scalable" isn't actually referenced anywhere. Is it supposed to be referenced in `color.scale()`?